### PR TITLE
*: fix global temp table will read tikv after schema change

### LINKED ISSUE #37306: Performance regression on BenchmarkInsertIntoSelect after pr/37225
## Enhancement

Ref https://github.com/pingcap/tidb/issues/25754
2022-08-22

image

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #37306

Problem Summary:

This bug is introduced by #37225.  When the information schema applying diff, it will create a builder and then call `InitWithOldInfoSchema` to copy old schema to the new one. However, the previous PR forgot to copy `temporaryTableIDs` in it.

### What is changed and how it works?

copy `temporaryTableIDs` in `InitWithOldInfoSchema`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by djshow832 — APPROVED** [review]

File: executor/temporary_table_test.go [hunk 1]
```diff
@@ -42,6 +42,14 @@ func TestTemporaryTableNoNetwork(t *testing.T) {
 		})
 	})
 
+	t.Run("global create and then create a normal table", func(t *testing.T) {
```
Comment (hawkingrei) [inline executor/temporary_table_test.go:45]: ```t.Run``` should be avoided. it is hard for bazel to speed up it by split into threads.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**